habitat/tasks/nav/object_nav_task.py

```diff
@@ -0,0 +1,149 @@
+        The object is visible from these positions.
+    """
+
+    object_id: str = attr.ib(default=None, validator=not_none_validator)
+    object_name: Optional[str] = None
+    object_category: Optional[str] = None
+    room_id: Optional[str] = None
+    room_name: Optional[str] = None
+    view_points: Optional[List[ObjectViewLocation]] = None
+
+
+@registry.register_sensor
+class ObjectGoalSensor(Sensor):
+    r"""A sensor for Object Goal specification as observations which is used in
+    ObjectGoal Navigation. The goal is expected to be specified by object_id or
+    semantic category id.
+    For the agent in simulator the forward direction is along negative-z.
+    In polar coordinate format the angle returned is azimuth to the goal.
+    Args:
+        sim: a reference to the simulator for calculating task observations.
+        config: a config for the ObjectGoalSensor sensor. Can contain field
+            GOAL_SPEC that specifies which id use for goal specification,
+            GOAL_SPEC_MAX_VAL the maximum object_id possible used for
+            observation space definition.
+        dataset: a Object Goal navigation dataset that contains dictionaries
+        of categories id to text mapping.
+    """
+
+    def __init__(
+        self, sim, config: Config, dataset: Dataset, *args: Any, **kwargs: Any
+    ):
+        self._sim = sim
+        self._dataset = dataset
+        super().__init__(config=config)
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "objectgoal"
+
+    def _get_sensor_type(self, *args: Any, **kwargs: Any):
+        return SensorTypes.SEMANTIC
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        sensor_shape = (1,)
+        max_value = (self.config.GOAL_SPEC_MAX_VAL - 1,)
+        if self.config.GOAL_SPEC == "TASK_CATEGORY_ID":
+            max_value = len(self._dataset.category_to_task_category_id)
+
+        return spaces.Box(
+            low=0, high=max_value, shape=sensor_shape, dtype=np.int64
+        )
+
+    def get_observation(
+        self,
+        observations,
+        *args: Any,
+        episode: NavigationEpisode,
+        **kwargs: Any,
+    ) -> Optional[int]:
+        if self.config.GOAL_SPEC == "TASK_CATEGORY_ID":
+            if len(episode.goals) == 0:
+                logger.error(
+                    f"No goal specified for episode {episode.episode_id}."
+                )
+                return None
+            if not isinstance(episode.goals[0], ObjectGoal):
+                logger.error(
+                    f"First goal should be ObjectGoal, episode {episode.episode_id}."
+                )
+                return None
+            category_name = episode.goals[0].object_category
+            return self._dataset.category_to_task_category_id[category_name]
+        elif self.config.GOAL_SPEC == "OBJECT_ID":
+            return np.array([episode.goals[0].object_name_id], dtype=np.int64)
+        else:
+            return None
```

Comment:
None will likely end up causing something else to crash and I don't see why you'd want the object goal sensor to sometimes not return a goal.  These `return None` should likely be `raise RuntimeError(<some error message>)` as that'll make debugging simpler.